pkg/sql/schemachanger/scplan/internal/rules/current/dep_create_function.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package current
+
+import (
+	"github.com/cockroachdb/cockroach/pkg/sql/schemachanger/rel"
+	"github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scpb"
+	. "github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scplan/internal/rules"
+	"github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scplan/internal/scgraph"
+)
+
+func init() {
+	registerDepRule(
+		"function existence precedes function dependents",
+		scgraph.Precedence,
+		"function", "dependent",
```

Comment:
Should this rule be more general and apply to all descriptors and their dependents?

Reply:
made it as a general rule